### ISSUE TITLE
Entries not being pushed to Loki correctly

### ISSUE THREAD

**Issue by oester** [BudgetSmartHome/home-lab-configs#3]
I do see data being pushed to Loki now, but several other errors are popping up:

    Sep 14 12:49:00 mon1 td-agent-bit[670]: [2021/09/14 12:49:00] [error] [parser] cannot parse 'Mgt1 7483c2086633,USW_FLEX_MINI-1.8.4.674: ENV:'
    Sep 14 12:49:00 mon1 td-agent-bit[670]: [2021/09/14 12:49:00] [ warn] [parser:syslog-rfc3164] invalid time format %b %d %H:%M:%S for 'Mgt1 7483c2086633,USW_FLEX_MINI-1.8.4.674: ENV:'
    Sep 14 12:49:02 mon1 td-agent-bit[670]: [2021/09/14 12:49:02] [error] [output:loki:loki.0] 192.168.15.71:3100, HTTP status=400 Not retrying.
    Sep 14 12:49:02 mon1 td-agent-bit[670]: entry with timestamp 2021-09-14 07:49:02 +0000 UTC ignored, reason: 'entry out of order' for stream: {city="null", country="null", job="fluentbit"},
    Sep 14 12:49:02 mon1 td-agent-bit[670]: total ignored: 1 out of 1
    Sep 14 12:49:07 mon1 td-agent-bit[670]: [2021/09/14 12:49:07] [error] [output:loki:loki.0] 192.168.15.71:3100, HTTP status=400 Not retrying.
    Sep 14 12:49:07 mon1 td-agent-bit[670]: entry with timestamp 2021-09-14 07:49:07 +0000 UTC ignored, reason: 'entry out of order' for stream: {city="null", country="null", job="fluentbit"},
    Sep 14 12:49:07 mon1 td-agent-bit[670]: total ignored: 1 out of 1

Basically, every syslog message is getting "entry out of order" or "invalid time format.

**Comment by proffalken** [comment]
Pretty sure this is related to #2 - try the fix there and see if it works, if not let me know and I'll revisit this.

I'll leave this open for now just in case.

**Comment by oester** [comment]
Removing the UTC filter from the config had no effect.Perhaps later version of the UniFi OS changed the syllogism format?

**Comment by proffalken** [comment]
Possibly, although I'm running the latest updates and I'm still seeing the logs without errors, leave it with me, I'll find time in the week to investigate, really appreciate you reporting this :)

**Comment by oester** [comment]
Actually, I think it's because the syslog messages from the UDM have a date stamp that includes the year:

Mon Sep 20 10:05:09 2021

Where as the format for RFC 3164 doesn't normally include the year:

Mon Sep 20 10:05:09 

Which matches up wit hthe error message:

invalid time format %b %d %H:%M:%S

**Comment by oester** [comment]
I pulled my previous comment, as it was a red herring. The "cannot parse date" error all from from the "unifi-flex-mini" switch which is clearly shooting out some invalid data, so let's ignore that for now.

I'm still seeing a fair number of "entry out of order" errors, not sure what's up there.

    Sep 20 16:06:42 mon1 td-agent-bit[14576]: [2021/09/20 16:06:42] [error] [output:loki:loki.0] 192.168.15.71:3100, HTTP status=400 Not retrying.
    Sep 20 16:06:42 mon1 td-agent-bit[14576]: entry with timestamp 2021-09-20 11:06:40 +0000 UTC ignored, reason: 'entry out of order' for stream: {city="null", country="null", job="fluentbit"},
    Sep 20 16:06:42 mon1 td-agent-bit[14576]: total ignored: 1 out of 1

**Comment by proffalken** [comment]
@oester - is Unifi the *only* thing sending logs to FluentBit, or have you configured some other devices too?

Those looks like like there's something on the network sending logs in UTC format, I'm wondering if something in your setup hasn't picked up the TZ settings you've given it somewhere?

**Comment by oester** [comment]
No other devices, just the UDM-pro. If I look at /var/log/messages on the UDM - it appears to be logging them in UTC, however the UDM date is local TZ:

    Sep 23 17:08:06 UDM-Pro daemon.info dnsmasq-dhcp[4201]: DHCPREQUEST(br25) 192.168.25.119 da:27:ac:8f:22:2a
    Sep 23 17:08:06 UDM-Pro daemon.info dnsmasq-dhcp[4201]: DHCPACK(br25) 192.168.25.119 da:27:ac:8f:22:2a 
    # date
    Thu Sep 23 12:10:24 CDT 2021

**Comment by oester** [comment]
Any follow-up to this? I came back to this recently, still have the "entry out of order" issue.

**Comment by proffalken** [comment]
Hey, sorry, I've not had a chance to really look in to it further, and I don't have a UDMP to test with either (I've got a USG3 and a CK2+ instead).

If I get a chance, I'll look again over the next few days, but without the ability to change timezones on my equipment without affecting the rest of my house, I'm not too sure how to test this further :(

**Comment by oester** [comment]
If you look at the error from fluent-bit:

    Oct 21 09:49:34 infra1 td-agent-bit[1343087]: [2021/10/21 09:49:34] [error] [output:loki:loki.0] 192.168.15.25:3100, HTTP status=400 Not retrying.
    Oct 21 09:49:34 infra1 td-agent-bit[1343087]: entry with timestamp 2021-10-21 09:49:23 +0000 UTC ignored, reason: 'entry out of order' for stream: {city="null", country="null", job="fluentbit"},
    Oct 21 09:49:34 infra1 td-agent-bit[1343087]: total ignored: 1 out of 1

Which is supposedly being parsed to a geolocation, it seems invalid. Everything is in my local TZ and I've removed the UTC conversion filter from fluent-bit. Looking at this error response from Loki, it does seem like a date/time issue tho... hmm. The timestamp says "2021-10-21 09:49:23 +0000 UTC" which is wrong, since it should have the UTC offset for my local TZ.

A syslog entry on my UDMP looks like this for instance (redacted a bit to hide my IP):

    Oct 21 10:13:27 UDM-Pro user.warn kernel: [760542.938871] IN=ppp0 OUT= MAC= SRC=89.248.165.247 DST=x.x.x.x LEN=40 TOS=0x00 PREC=0x00 TTL=242 ID=30077 PROTO=TCP SPT=59865 DPT=62851 WINDOW=1024 RES=0x00 SYN URGP=0
